test/mpi/rma/getfence1.c

```diff
@@ -9,52 +9,60 @@
 #include <string.h>
 #include <limits.h>
 #include "mpitest.h"
-
-#define LARGE_CNT_CONTIG    550000000
-#define LARGE_CNT_NONCONTIG 150000000
+#include "dtpools.h"
 
 /*
 static char MTEST_Descrip[] = "Get with Fence";
 */
 
 static inline int test(MPI_Comm comm, int rank, int source, int dest,
-                       MTestDatatype * sendtype, MTestDatatype * recvtype)
+                       int i, int j, DTP_t send, DTP_t recv)
 {
     int errs = 0, err;
     int disp_unit;
+    int len;
+    MPI_Aint sendcount, recvcount, count;
     MPI_Aint extent, lb;
     MPI_Win win;
+    MPI_Datatype sendtype, recvtype;
+    char send_name[128] = { 0 };
+    char recv_name[128] = { 0 };
+    void *sendbuf, *recvbuf;
+
+    count = send->DTP_type_signature.DTP_pool_basic.DTP_basic_type_count;
+    sendtype = send->DTP_obj_array[i].DTP_obj_type;
+    recvtype = recv->DTP_obj_array[j].DTP_obj_type;
+    sendcount = send->DTP_obj_array[i].DTP_obj_count;
+    recvcount = recv->DTP_obj_array[j].DTP_obj_count;
+    sendbuf = send->DTP_obj_array[i].DTP_obj_buf;
+    recvbuf = recv->DTP_obj_array[j].DTP_obj_buf;
+
+    MPI_Type_get_name(sendtype, send_name, &len);
+    MPI_Type_get_name(recvtype, recv_name, &len);
 
     MTestPrintfMsg(1,
                    "Getting count = %ld of sendtype %s - count = %ld receive type %s\n",
-                   sendtype->count, MTestGetDatatypeName(sendtype), recvtype->count,
-                   MTestGetDatatypeName(recvtype));
-    /* Make sure that everyone has a recv buffer */
-    recvtype->InitBuf(recvtype);
-    sendtype->InitBuf(sendtype);
-    /* By default, print information about errors */
-    recvtype->printErrors = 1;
-    sendtype->printErrors = 1;
-
-    MPI_Type_extent(sendtype->datatype, &extent);
-    MPI_Type_lb(sendtype->datatype, &lb);
+                   sendcount, send_name, recvcount, recv_name);
+
+    MPI_Type_extent(sendtype, &extent);
+    MPI_Type_lb(sendtype, &lb);
     disp_unit = extent < INT_MAX ? extent : 1;
-    MPI_Win_create(sendtype->buf, sendtype->count * extent + lb, disp_unit, MPI_INFO_NULL, comm,
-                   &win);
+    MPI_Win_create(sendbuf, sendcount * extent + lb, disp_unit, MPI_INFO_NULL, comm, &win);
     MPI_Win_fence(0, win);
+
+    /* To improve reporting of problems about operations, we
+     * change the error handler to errors return */
+    MPI_Win_set_errhandler(win, MPI_ERRORS_RETURN);
+
```

Comment:
The code does not handle error on source process. You might want to move the `set_errhandler` back.

Reply:
Can you clarify this?